src/application/db/__init__.py

```diff
@@ -0,0 +1,18 @@
+from injector import Module, provider, singleton
+from sqlalchemy import create_engine
+from sqlalchemy.engine import Engine
+from sqlalchemy.orm import declarative_base
+
+from application.settings import Settings
+
+Base = declarative_base()
+
+
+class DBModule(Module):
+    @provider
+    @singleton
+    def engine(self, settings: Settings) -> Engine:
+        return create_engine(settings.database_url, echo=False, future=True)
```

Comment:
I don't think that passing the entire `Settings` instance is a good idea as it makes suddenly all modules (and sometimes even classes!) depend on it

Reply:
But this is Injector Module here.
Is there a way to pass only attribute?

Reply:
> Is there a way to pass only attribute?
Nope

Here's a pattern I use:

```python
from attr import define
from injector import provider, singleton

@define
class DBModule(Module):
    _url: str
 
    @singleton
    @provider
    def engine(self) -> Engine:
        return create_engine(self._url, echo=False, future=True)
```

Then you only pass an attribute of settings when you instantiate `DBModule` and create the final `Injector` instance.

This solution makes fewer objects depending on settings. In my project, `Settings` is an ephemeral object that's lives very shortly - when the program starts it is created by reading configuration and then garbage-collected after an Injector instance is created.

Reply:
good point. :+1: 
changed